src/app/[locale]/onboarding/loading/_hooks/use-logic.ts

```diff
@@ -0,0 +1,180 @@
+"use client";
+
+import { useQuery } from "@tanstack/react-query";
+import i18next from "i18next";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+import { DEFAULT_PACKAGE_API_URL } from "~/constants";
+import usePackageAPI from "~/hooks/use-package-api";
+import useSQLInit from "~/hooks/use-sql-init";
+import { useAppStore } from "~/stores";
+
+type API = ReturnType<typeof usePackageAPI>;
+
+function useURLParams() {
+  const searchParams = useSearchParams();
+
+  const packageLink = searchParams?.get("packageLink") || undefined;
```

Comment:
```suggestion
  const packageLink = searchParams?.get("packageLink") ?? undefined;
```

Use of `??` would be more appropriate in this situation

Reply:
https://caniuse.com/mdn-javascript_operators_logical_or has better support than https://caniuse.com/mdn-javascript_operators_nullish_coalescing and it's achieving more or less the same in this case